### PR TITLE
Update a11y attributes for consistency

### PR DESCRIPTION
**Pull request by yomed**
### Breadcrumb
- `heading-level` -> `a11y-heading-tag`
- `heading-text` -> `a11y-heading-text`

### Carousel
- `accessibility-prev` ->`a11y-previous-text`
- `accessibility-next` ->`a11y-next-text`
- `accessibility-status` ->`a11y-status-text`
- `accessibility-current` ->`a11y-current-text`
- `accessibility-other` ->`a11y-other-text`
- `accessibility-pause` ->`a11y-pause-text`
- `accessibility-play` ->`a11y-play-text`

### Combobox
- `label` -> `text` (for option tag)

### Dialog
- `aria-label-close` -> `a11y-close-text`

### Icon
- `accessibility-text` -> `a11y-text`

### Menu
- `label` -> `text`
- `accessibility-text` -> `a11y-text`

### Notice
- `heading-level` -> `a11y-heading-tag`
- `aria-text` -> `a11y-heading-text`
- `aria-label-close` -> `a11y-close-text`

### Pagination
- `accessibility-prev` -> `a11y-previous-text`
- `accessibility-next` -> `a11y-next-text`
- `accessibility-current` -> `a11y-current-text`

### Select
- `label` -> `text` (for item tag)

## Context

Accessibility attributes have had lots of different names throughout development so far, so now we've settled on a basic standard. In general, I cleaned up all references to `accessibility` or `label`.

## References

#214

### CODE REVIEW

File: src/components/ebay-menu/test/test.server.js [hunk 1]
```diff
@@ -104,10 +104,10 @@ describe('menu', () => {
         expect($('.expand-btn.expand-btn--primary').length).to.equal(1);
     });
 
-    test('renders without label', context => {
-        const input = { label: '' };
+    test('renders without textl', context => {
```
Comment (seangates) [inline src/components/ebay-menu/test/test.server.js]: Minor: `textl` should be `text`

File: src/components/ebay-breadcrumb/index.js [hunk 2]
```diff
@@ -40,8 +40,8 @@ function getTemplateData(state, input) {
         style: input.style,
         items,
         hijax,
-        headingText: input.headingText || '',
-        headingLevel: input.headingLevel || 'h2'
+        a11yHeadingText: input.a11yHeadingText || '',
+        a11yHeadingLevel: input.a11yHeadingLevel || 'h2'
```
[the remark this answers is not in the document]
Comment (yomed) [reply src/components/ebay-breadcrumb/index.js]: Yeah, I saw this. I'm thinking we call it `a11y-heading-tag` and then have users pass in the full tag like `h2`. Supplying an integer divorced from the HTML tag just seems weird. But, `level` implies an integer, hence the name change.

[the remark this answers is not in the document]
Comment (seangates) [reply src/components/ebay-breadcrumb/index.js]: Let's go with `ally-heading-tag` everywhere. Developers will then know better what to do.

**Comment by mlrawlings** [comment]
To make the upgrade smoother, it might be worth targeting this change against one of the `0.x` releases and adding some transformers that will compile the old attributes into the new attributes:

```js
const attrMap = {
  "accessibility-prev": "a11y-previous-text",
  "accessibility-next": "a11y-next-text",
  "accessibility-status": "a11y-status-text",
  "accessibility-current": "a11y-current-text",
  "accessibility-other": "a11y-other-text",
  "accessibility-pause": "a11y-pause-text",
  "accessibility-play": "a11y-play-text",
};

module.exports = (el, context) => {
  Object.entries(attrMap).forEach(([oldName, newName]) => {
    if (el.hasAttribute(oldName) {
      if (context.deprecate) {
        // marko 4 has deprecation warning support
        // this will output the line number where the old attribute name was used
        context.deprecate(`The ${oldName} attribute has been renamed to ${newName}`);
      } else {
        // generic warning for marko 3
        console.warn(`The ${oldName} attribute has been renamed to ${newName}`);
      }

      var value = el.getAttributeValue(oldName);
      el.setAttributeValue(newName, value);
      el.removeAttribute(oldName);
    }
  });
};
```

**Comment by seangates** [comment]
@mlrawlings We actually waited to do this until we knew we'd have breaking changes. Hence, doing it in v1.0.0 instead of v0.8.x or v0.9.x.

**Comment by seangates** [comment]
Although, I do like the idea of "deprecation" for our stuff. It just makes things ... heavier.

**Comment by yomed** [comment]
Deprecation is certainly good for users, though I also wouldn't want to overdo it in terms of our own burden. It does get a little messy for us with the v3/v4 split. The warning itself is easy to split like above, but only v4 supports multiple transformers (AFAIK), and a bunch of our components have existing transformers. We should probably come up with general guidelines around how we want to handle deprecation.

**Comment by msendlakowski** [comment]
Yeah, I think we're OK before v1.  Applications should expect some breaking changes.  But I'll create an issue for discussion in the backlog regarding deprecation in general.

**Comment by msendlakowski** [comment]
@seangates  have your requested changes been addressed?

**Comment by yomed** [comment]
@seangates All good on this one?

**Comment by seangates** [comment]
Sorry about the delay!

**Pull request merged**